docs/polygon_draw.js

```diff
@@ -366,22 +387,79 @@ function setUpPolygonDrawing(map, firebasePromise) {
+ *     (only used by tests). Null if there was an error
  */
-function initializeAndProcessUserRegions(map, firebasePromise) {
+async function initializeAndProcessUserRegions(map, firebasePromise) {
   addLoadingElement(mapContainerId);
-  return firebasePromise
-      .then((doc) => {
-        // Damage asset may not exist yet, so this is undefined. We tolerate
-        // gracefully.
-        damageAsset = doc.data()['asset_data']['damage_asset_path'];
-        userShapes = getFirestoreRoot().collection(collectionName);
-      })
-      .then(() => userShapes.get())
-      .then(
-          (querySnapshot) => drawRegionsFromFirestoreQuery(querySnapshot, map))
-      .catch(createError('getting user-drawn regions'));
+  try {
+    setUpPopup();
+    let doc;
+    try {
+      doc = await firebasePromise;
+    } catch (err) {
+      // Firebase retrieval error handled elsewhere.
+      throw err;
+    }
+    // Damage asset may not exist yet, so this is undefined. We tolerate
+    // gracefully.
+    damageAsset = doc.data()['asset_data']['damage_asset_path'];
+    userShapes = getFirestoreRoot().collection(collectionName);
+    let querySnapshot;
+    try {
+      querySnapshot = await userShapes.get();
+    } catch (err) {
+      handleUserShapesError(err);
+      return null;
+    }
+    drawRegionsFromFirestoreQuery(querySnapshot, map);
+    return setUpPolygonDrawing(map);
+  } finally {
+    loadingElementFinished(mapContainerId);
+  }
+}
+
+const AUTHENTICATION_ERROR_CODE = 'permission-denied';
+const USER_FEATURES_DIALOG =
+    '<div>Sign in to authorized account to view user-drawn features</div>';
+
+/**
+ * Handles errors thrown from retrieving user-defined features, most likely due
+ * to authentication issues.
+ * @param {Error|firebase.FirebaseError} err
+ */
+function handleUserShapesError(err) {
+  const userFeaturesRow = $('#' + userFeaturesCheckboxRowId);
+  userFeaturesRow.css('text-decoration', 'line-through');
+  // Checkbox just causes issues with double event triggering, and is useless
+  // anyway. Get rid of it.
+  userFeaturesRow.children('input').remove();
+  if (err.code === AUTHENTICATION_ERROR_CODE) {
+    showError('Viewing as public, private data not available', null);
+    userFeaturesRow.prop(
+        'title',
+        'User features not available: please click to log in to ' +
+            'authorized account');
+    const popUpDialog = () => {
+      const dialogParent = $(USER_FEATURES_DIALOG);
+      const dialog = dialogParent.dialog({
+        buttons: [
+          {text: 'Sign in', click: reloadWithSignIn},
+          {text: 'Close', click: () => dialog.dialog('close')},
+        ],
+        modal: true,
+        width: 600,
+        close: () => {
+          // Don't try to reuse the dialog.
+          dialogParent.empty();
+          dialogParent.remove();
+        },
+      });
+    };
+    userFeaturesRow.on('click', popUpDialog);
```

Comment:
would it be possible to add a ? help icon like we're using elsewhere? Think it would go a long way to making it clear you can get help for whatever is happening here

Reply:
Nice idea, done.

Reply:
@ruthtalbot fyi gave you a TODO here